### PR TITLE
MIR-943 remove support of @display=false for access checks

### PR DESCRIPTION
**Pull request by yagee-de**
a derivate should use mir_access as a classification to restrict access

[Link to jira](https://mycore.atlassian.net/browse/MIR-943).

### CODE REVIEW

**Comment by yagee-de** [comment]
Still missing: GUI and migration command

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=6223826&bid=20285689)

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=6223826&bid=20285689)

File: mir-module/src/main/java/org/mycore/mir/migration/MIRMigration202006Utils.java [hunk 1]
```diff
@@ -39,13 +44,58 @@
...
+    public static List<String> migrateDerivateDisplay(String targetCategory) {
+        MCRCategoryID categoryID = MCRCategoryID.fromString(targetCategory);
+        if (!MCRCategoryDAOFactory.getInstance().exist(categoryID)) {
+            throw new MCRException("Category " + categoryID + " does not exist!");
+        }
+        final MCRXMLMetadataManager mcrxmlMetadataManager = MCRXMLMetadataManager.instance();
+        final List<String> derivates = mcrxmlMetadataManager.listIDsOfType("derivate");
+        return derivates.stream()
+            .map(MCRObjectID::getInstance)
+            .map(mcrid -> {
```
Comment (sebhofmann) [inline mir-module/src/main/java/org/mycore/mir/migration/MIRMigration202006Utils.java:68]: This always looks ugly, maybe its worth to include apache-commons-lang3. You could then do this:
`Functions.apply(mcrxmlMetadataManager::retrieveXML, mcrid)`

There are other cases alone in this file where we could use this.

**Review by sebhofmann — COMMENTED** [review]
Looks fine for me.

File: mir-module/src/main/java/org/mycore/mir/migration/MIRMigration202006Utils.java (hunk 1 above)
Comment (sebhofmann) [reply mir-module/src/main/java/org/mycore/mir/migration/MIRMigration202006Utils.java:68]: http://commons.apache.org/proper/commons-lang/javadocs/api-3.9/index.html?org/apache/commons/lang3/Functions.html

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=6223826&bid=20285689)

**Review by kkrebs — APPROVED** [review]

**Comment by yagee-de** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity decreasing per file
==============================
+ mir-module/src/main/java/org/mycore/mir/authorization/MIRStrategy.java  -3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/MyCoRe/mir/pullRequest?prid=6223826&bid=20285689)

**Pull request merged**